### PR TITLE
Wg client swc

### PR DESCRIPTION
**Pull request by Cedran-Bastien**
### Goal: 
- Get SWC file needed to complile flash 

### Done: 
- Add flash game lib builder 
-> get the archive containing SWC file, unzip them and put it in .wg-mod/flash_lib

### To_test:
Feature is not actually executed. it will be during flash build 
If you want to test, you will have to add this line: 
`config/mod.rs -> Configs::load() `
in file:
`build_flash_client_lib(&wg_mod_home)?;`

### Question : 
- When do we launch the build ?
only on flash build ? all the time on execution(if the directory don't already exist) ? other better idea ?

### TODO:
- manage error case when the `game_client` directory (in settings) exist but not contain the wot-client ( [Corresponding  #issue](https://github.com/gabrielhamel/wg-mod/issues/42)) : 

In that case the settings will be created because directory exist and the builder throw an error because archive are not found.
the problem is that the user can't change game_client_path settings without knowing the utils and delete the file `.../.wg-mod/settings.json`
SOLUTION: add a CLI command to change the game_client_path settings and put a message explaining that instead of throwing error

### CODE REVIEW

File: src/builder/flash_lib.rs [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+    fn is_present(&self) -> bool {
+        self.game_flash_lib.exists()
+    }
+
+    fn build(&self) -> Result<(), Error> {
+        let tmp_dir =
+            tempdir().map_err(|e| Error::BuildError(e.to_string()))?;
+
+        let archive_indentifier = vec!["1", "2", "3"];
```
Comment (gabrielhamel) [inline src/builder/flash_lib.rs]: not very evolutive, try to list file in archive and get all gui_part_X

File: src/utils/extract_archive.rs [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+use zip::write::{ExtendedFileOptions, FileOptions};
+use zip::{CompressionMethod, ZipArchive, ZipWriter};
+
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    #[error("Archive does not exist: {0}")]
+    ArchiveError(String),
+    #[error("Failed to extract archive: {0}")]
+    ExtractError(String),
+    #[error("Failed to extract archive: {0}qqqqqzzzzz")]
```
Comment (gabrielhamel) [inline src/utils/extract_archive.rs]: ?

File: src/utils/extract_archive.rs [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+fn mock_archive(path: &PathBuf, archive_name: String) {
+    let file = File::create(path.join(archive_name)).unwrap();
+    let writer = BufWriter::new(file);
+    let mut zip = ZipWriter::new(writer);
+    let options: FileOptions<'_, ExtendedFileOptions> =
+        FileOptions::default().compression_method(CompressionMethod::Deflated);
+
+    zip.start_file("build_test.txt", options).unwrap();
+    zip.write_all(b"Hello, world!").unwrap();
```
Comment (gabrielhamel) [inline src/utils/extract_archive.rs:63]: attended ?

Comment (Cedran-Bastien) [reply src/utils/extract_archive.rs:63]: the unwrap statement ?
yes it's just for testing

**Pull request merged**